### PR TITLE
Fix kernel selftest [v2]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
v1: #233 
v2:
1-removed variable name as udeps,rdeps,sdeps,cdeps 
    extend base packages list (deps) required distro packages

2- use variable name count instead of self.count as boolean variable 
    rectified  message string

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
job log : 
[joblog.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/711637/joblog.txt)

File: kernel/kselftest.py [hunk 1]
```diff
@@ -46,22 +47,19 @@ def setUp(self):
...
-        udeps = ['git', 'popt', 'build-essential',
-                 'libpopt-dev', 'libpopt0', 'libcap-dev', 'libcap-ng-dev']
-        rdeps = ['git', 'popt', 'popt-static', 'glibc', 'glibc-devel',
-                 'glibc-static', 'libcap-ng', 'libcap-ng-devel', 'libcap1', 'libcap1-devel']
-        cdeps = ['git', 'popt', 'popt-static', 'glibc', 'glibc-devel',
-                 'glibc-static', 'libcap-ng', 'libcap-ng-devel', 'libcap', 'libcap-devel']
         if 'Ubuntu' in detected_distro.name:
-            deps = deps + udeps
+            deps.extend(['git-core', 'popt', 'glibc', 'glibc-devel', 'popt-devel',
+                         'libcap1', 'libcap1-devel', 'libcap-ng', 'libcap-ng-devel'])
```
Comment (apahim) [inline kernel/kselftest.py]: Ok, this is much more clear.

File: kernel/kselftest.py [hunk 2]
```diff
@@ -46,22 +47,19 @@ def setUp(self):
...
         elif 'redhat' in detected_distro.name:
-            deps = deps + rdeps
+            deps.extend(['git', 'popt', 'popt-static', 'glibc', 'glibc-devel', 'glibc-static',
+                         'libcap-ng', 'libcap-ng-devel', 'libcap1', 'libcap1-devel'])
         elif 'SuSE' in detected_distro.name:
-            deps = deps + sdeps
-        elif 'centos' in detected_distro.name:
-            deps = deps + cdeps
+            deps = deps.extend(['git-core', 'popt', 'glibc', 'glibc-devel', 'popt-devel',
```
Comment (apahim) [inline kernel/kselftest.py]: Leftover? (`deps = `)

File: kernel/kselftest.py [hunk 3]
```diff
@@ -92,10 +90,21 @@ def test(self):
         """
         Execute the kernel selftest
         """
+        count = False
```
Comment (apahim) [inline kernel/kselftest.py]: This is not a counter anymore, that's why I suggested the name of this variable to be `errors`.

File: kernel/kselftest.py (hunk 2 above)
Comment (PraveenPenguin) [inline kernel/kselftest.py]: oh my bad luck will do

**Comment by PraveenPenguin** [comment]
@apahim pushed the changes can you please have a look

**Pull request merged**